code/modules/ruins/objects_and_mobs/necropolis_gate.dm

```diff
@@ -236,17 +240,70 @@ GLOBAL_DATUM(necropolis_gate, /obj/structure/necropolis_gate/legion_gate)
 	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | UNACIDABLE | ACID_PROOF
 	var/tile_key = "pristine_tile"
 	var/tile_random_sprite_max = 24
+	var/fall_on_cross = STABLE //If the tile has some sort of effect when crossed
+	var/fallen = FALSE //If the tile is unusable
+	var/falling = FALSE //If the tile is falling
 
 /obj/structure/stone_tile/Initialize(mapload)
 	. = ..()
 	icon_state = "[tile_key][rand(1, tile_random_sprite_max)]"
 
 /obj/structure/stone_tile/Destroy(force)
-	if(force)
+	if(force || fallen)
 		. = ..()
 	else
 		return QDEL_HINT_LETMELIVE
 
+/obj/structure/stone_tile/Crossed(atom/movable/AM)
+	var/turf/T = get_turf(src)
+	if(!istype(T, /turf/open/floor/plating/lava) && !istype(T, /turf/open/chasm)) //nothing to sink or fall into
+		return
+	if(falling || fallen)
+		return
+	var/obj/item/I
+	if(istype(AM, /obj/item))
+		I = AM
+	var/mob/living/L = AM
+	if(isliving(AM))
+		L = AM
```

Comment:
If there is an issue here, I'm not aware of it. Why is this bad? What can I do to fix it? I can do nothing if you do not tell me the problem.